File: server/src/main/java/com/bomberman/factories/PlayerFactory.java
package com.bomberman.factories;

import com.bomberman.constants.Configuration;
import com.bomberman.entities.Position;
import com.bomberman.entities.mapobject.movable.Player;
import com.bomberman.game.GameState;

public class PlayerFactory {

  public Player createPlayer(final String id, final GameState gameState) {
    return new Player(
        id,
        new Position(0, Configuration.MAP_HEIGHT - Configuration.PLAYER_TEXTURE_HEIGHT),
        gameState);
  }
}

File: server/src/main/kotlin/com/bomberman/factory/PlayerFactory.kt
package com.bomberman.factory

import com.bomberman.MAP_HEIGHT
import com.bomberman.PLAYER_TEXTURE_HEIGHT
import com.bomberman.entity.Position
import com.bomberman.entity.mapobject.movable.Player
import com.bomberman.game.GameState

class PlayerFactory {

    fun createPlayer(id: String, gameState: GameState): Player =
            Player(
                    id = id,
                    position = Position(0, MAP_HEIGHT - PLAYER_TEXTURE_HEIGHT),
                    gameState = gameState)
}